### DISCUSSION TITLE
How can I host over local network?

### DISCUSSION THREAD

**Discussion by peteydidit** [lllyasviel/Fooocus#1333] [Q&A] [answered]
Right now the server is locked to localhost, not accessible by the machine's IP address over local network. I have a machine on my network with a better GPU that I would rather use for this. Can I change some params or config files to spin up a server? Perhaps there's even a simple code change I could do.

**Comment by ItsJoshBrown** [comment, marked as answer]
@peteydidit 
Just a user so take it with a grain of salt. 

I can't say this is the correct way, but I had to keep experimenting until I was able to figure it out. What I ended up doing on my machine since I was set up similar to what you have I made a copy of one of the  .bat files and changed it to this. 

as an example, I changed my run_realistic.bat

Originally it was set to this. 

```
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --preset realistic
pause
```
I modified it to look like this. 
```
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --preset realistic --listen 192.168.X.X --port 7865
pause
```

changing 192.168.X.X  to whatever IP your local machine is on.


You will also have to allow it through the firewall, so do so at your own risk. I still haven't dived that far into it to see what is what and where.

**Reply by rsdpx** [reply]
however, it's work！ thx

**Reply by mashb1t** [reply]
you can just add --listen to your run.bat to listen on all IP addresses!

**Reply by Wontfallo** [reply]
What needs to be let through the firewall???? Python, Brave??  This isn't working for me and I can't understand why.   I've tried with  --listen, and also --listen [ip ad+

drees of local machine running fooocus]  and also --listen --port 8000  and all the variations of this. I can hope it in a vm on the same machine, but I cant open it on any other devices that are connected to my network. What could I be missing here? please help,

**Reply by Rogala** [reply]
The `--listen` parameter starts Fooocus with access through any network interface on your PC with port `:7865`
```
127.0.0.1:7865
localhost:7865
local-ip:7865
external-ip:7865
```
etc...

To specify the network address and port explicitly, use the parameter
`--listen xxx.xxx.xxx.xxx.xxx --port xxxx
`
To open network access you need to allow connection in the firewall to the IP:port on the local network of your PC running the Fooocus instance.

**Reply by mashb1t** [reply]
@Wontfallo in addition to this you need to check if you have enabled network separation by device and your devices can communicate in general with each other. Most "guest" WiFi networks have this option enabled, so please double check to rule out this factor.

**Reply by Jayant-1** [reply]
It works

**Reply by Identity9165** [reply]
> you can just add --listen to your run.bat to listen on all IP addresses!

thanks. this is how my setup become available on tailscale.

**Comment by shahafbenshaloom** [comment]
i dont undestand what do i need to do after.

i changed the .bat

i open the port

(dont have a fire wall on that server) went to the other computer and tried:

server ip
server ip:port

server ip/127.0.0.1:port

none worked. what am i missing?

**Reply by Rogala** [reply]
This is enough to allow you to access your server on the local network using its IP and the standard port 7865
`.\python_embeded\python.exe -s Fooocus\entry_with_update.py --listen`

**Reply by shahafbenshaloom** [reply]
ok sorry i miss read aht you wrote. i need to check my connection i guess, perhaps there is some kind of firewall i missed.

**Reply by Rogala** [reply]
Ping your computer with the server to see if it is available on the network

**Reply by shahafbenshaloom** [reply]
ok it was set on public... its working now!
have any idea where are the outputs? on the server? or can i set for them to be in the computer i'm working on?
